src/NUnitEngine/nunit.engine/Services/RuntimeFrameworkService.cs

```diff
@@ -96,7 +96,10 @@ private static bool FrameworksMatch(RuntimeFramework f1, RuntimeFramework f2)
         /// <returns>A string representing the selected RuntimeFramework</returns>
         public string SelectRuntimeFramework(TestPackage package)
         {
-            // Start by examining the provided settings
+            // Evaluate package target framework
+            ApplyImageData(package);
+
```

Comment:
Where `ApplyImageData` is done should have no effect - I just moved it up to the top, so it was out of the way.